### PR TITLE
Fixes #8121: Clear package caches when repairing a package repository

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8121

### CODE REVIEW

**Comment by peckpeck** [comment]
This looks like a bug and not a user story, moreover the issue is missing a description of a reason of this change, what problem does it fix ?

**Comment by amousset** [comment]
I updated the issue description.

File: techniques/applications/zypperPackageManagerRepositories/1.0/zypper-repositories-management.st [hunk 1]
```diff
@@ -75,6 +75,14 @@ bundle agent zypper_repositories_management
     SuSE::
       "repos_${zypper_index}" usebundle => rudder_common_reports_generic("Zypper Repositories Management", "zypper_repo_${zypper_index}", "${zypper_uuid[${zypper_index}]}", "Repositories", "${zypper_name[${zypper_index}]}", "Repository ${zypper_name[${zypper_index}]}");
 
+      # Clear package caches
+      "remove_caches" usebundle => file_remove("/var/rudder/cfengine-community/state/software_packages.csv"),
```
Comment (peckpeck) [inline techniques/applications/zypperPackageManagerRepositories/1.0/zypper-repositories-management.st]: Doesn't this contain the installed package list ?
If so, why would you need to update it after a repository change ?

Comment (amousset) [inline techniques/applications/zypperPackageManagerRepositories/1.0/zypper-repositories-management.st]: These files are created by the agent when they do not exist.

Comment (amousset) [inline techniques/applications/zypperPackageManagerRepositories/1.0/zypper-repositories-management.st]: We do not need to remove, this file, indeed.

**Comment by amousset** [comment]
PR updated

**Comment by peckpeck** [comment]
OK, merging

**Pull request merged**